drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -151,6 +181,47 @@ static inline void msg_init(const struct device *dev, struct i2c_msg *msg,
 #if defined(CONFIG_I2C_TARGET)
 		data->master_active = true;
 #endif
+
+#ifdef I2C_STM32_V2_DMA
+		if (msg->len) {
+			if (msg->flags & I2C_MSG_READ) {
+				/* Configure RX DMA */
+				k_sem_take(&data->dma_rx_sem, K_FOREVER);
+				data->dma_rx_blk_cfg.source_address = LL_I2C_DMA_GetRegAddr(
+					cfg->i2c, LL_I2C_DMA_REG_DATA_RECEIVE);
+				data->dma_rx_blk_cfg.dest_address = (uint32_t)msg->buf;
+				data->dma_rx_blk_cfg.block_size = msg->len;
+
+				if (configure_dma(&cfg->rx_dma, &data->dma_rx_cfg,
+						  &data->dma_rx_blk_cfg) != 0) {
+					k_sem_give(&data->dma_rx_sem);
+					return;
```

Comment:
If we return here, i guess what will happend is that waiting thread will timeout
Maybe that is intended, but in that case it would be good to add a LOG_WRN() before returning here

Reply:
Done, I wonder if we want to change this function signature to return an error code

Reply:
Yes i agree it would be good to return with an error code if configure_dma fails

Reply:
I'm happy to do it, but I wonder to keep this MR short as possible to do that in a follow up piece of work. @erwango has mentioned keeping changes to a minimum due to other proposed pieces of work going on

Reply:
I understand, then just a log message will do for now